test/onemkl.jl

```diff
@@ -807,5 +813,34 @@ end
                 @test C ≈ h_C
             end
         end
+        @testset "gemm!" begin
```

Comment:
It seems that we will need to add Float16, ComplexF16 to the GEMM test similar to the copy test.

Reply:
Thanks for the review, yes, it is added now.

Reply:
Looks good and thank you!